### PR TITLE
Cache Room Flags

### PR DESCRIPTION
**Pull request by Spedwards**
This will save some CPU when searching for local flags.

Also did a little reformatting to clean the syntax.

### CODE REVIEW

File: flagDir.js [hunk 1]
```diff
@@ -11,51 +11,56 @@ mod.flagFilter = function(flagColour) {
...
+    let list = this.list;
+    if (flagColor === null || list.length === 0) return null;
     let filter;
     if (pos instanceof Room) pos = pos.getPositionAt(25, 25);
     if (typeof flagColor === 'function' ) {
         filter = function(flagEntry) {
-            if ( flagColor(flagEntry) && flagEntry.cloaking == 0 ) {
-                if ( !local ) return true;
-                if ( pos && pos.roomName && flagEntry.roomName == pos.roomName ) return true;
+            if (flagColor(flagEntry) && flagEntry.cloaking === 0) {
```
Comment (karlthepagan) [inline flagDir.js]: `!flagEntry.cloaking` otherwise this will have unintended logical changes

File: flagDir.js [hunk 2]
```diff
@@ -69,40 +74,51 @@ mod.removeFromDir = function(name){
...
+        } else {
+            _.assign(filter, {roomName: pos.roomName});
+        }
+    }
+    return _.countBy(list, filter).true || 0;
 };
 mod.filter = function(flagColor, pos, local=true){
-    if( flagColor == null || this.list.length == 0)
-        return [];
+    if (flagColor === null || this.list.length === 0) return [];
```
Comment (karlthepagan) [inline flagDir.js]: `!flagColor`

File: flagDir.js [hunk 3]
```diff
@@ -69,40 +74,51 @@ mod.removeFromDir = function(name){
...
+            }
+        }
     }
-    return _.filter(this.list, filter);
+    return _.filter(list, filter);
 };
-mod.filterCustom = function(filter){
-    if( filter == null || this.list.length == 0)
+mod.filterCustom = function(filter) {
+    if (filter =9== null || this.list.length === 0)
```
Comment (karlthepagan) [inline flagDir.js]: `!filter`

Comment (Spedwards) [inline flagDir.js]: There's also a `9` in there somehow...

**Pull request merged**